ark/type/parser/objectLiteral.ts

```diff
@@ -24,8 +28,44 @@ const stringAndSymbolicEntriesOf = (o: Record<string | symbol, unknown>) => [
 export const parseObjectLiteral = (def: Dict, ctx: ParseContext) => {
 	const required: Inner<"required">[] = []
 	const optional: Inner<"optional">[] = []
+
+	// We only allow a spread operator to be used as the first key in an object
+	// because to match JS behavior any keys before the spread are overwritten
+	// by the values in the target object, so there'd be no useful purpose in having it
+	// anywhere except for the beginning.
+	// Discussion in ArkType Discord:
+	// https://discord.com/channels/957797212103016458/1103023445035462678/1182814502471860334
+	let hasSeenFirstKey = false
+
 	for (const entry of stringAndSymbolicEntriesOf(def)) {
 		const result = parseEntry(entry)
+
+		if (result.kind === "spread" && hasSeenFirstKey) {
+			return throwParseError(
+				"Spread operator may only be used as the first key in an object"
+			)
+		} else if (result.kind === "spread") {
+			const spreadNode = ctx.scope.parse(result.innerValue, ctx)
+
+			if (!spreadNode.extends(keywords.object)) {
+				return throwParseError(
+					writeInvalidSpreadTypeMessage(printable(result.innerValue))
+				)
+			}
+
+			if (spreadNode.kind !== "intersection") {
```

Comment:
I would have a single check consisting of the first check you made against object which is perfect.

I'd have to think more about whether we eventually want to allow something like `"...": "Array"`, although it doesn't seem like a great idea to recheck all those builtin methods as opposed to just checking the prototype.

For now, just checking if it's an intersection seems fine since that's the only type that can actually have unique props on it.

So anyways, probably just make it one message like `Spread operand must resolve to an object literal type (was `${printable(result.innerValue))}`` or similar.

Reply:
Okay, so `T|U` will not be a node that extends object? There's a type error if I don't do that.

Reply:
No sorry I meant combining the two checks like `spreadNode.kind !== "intersection" || !spreadNode.extends(keywords.object)` with that one error message

Reply:
Ah, understood, thank you!

Reply:
e83da76